File: in-motion-server/Friends/IMS.Friends.Models/Dto/Outgoing/InvitationFriendshipDto.cs
namespace IMS.Friends.Models.Dto.Outgoing;

public class InvitationFriendshipDto
{
    public string Id { get; set; }
    
    public string ExternalUserId { get; set; }
    
    public FriendInfoDto ExternalUser { get; set; }
    
    public DateTime Invited { get; set; }
}

File: in-motion-android/app/src/main/java/com/inmotion/in_motion_android/data/remote/dto/friends/InvitationFriendshipDto.kt
package com.inmotion.in_motion_android.data.remote.dto.friends

import com.inmotion.in_motion_android.data.database.entity.InvitedFriend

data class InvitationFriendshipDto(
    val id: String,
    val externalUserId: String,
    val externalUser: FriendInfoDto,
    val invited: String
) {
    fun toInvitedFriend(): InvitedFriend {
        return InvitedFriend(
            externalUserId,
            id,
            externalUser.nickname,
            externalUser.bio,
            invited
        )
    }
}
